### PR TITLE
docs: Fix some typos and markdown links

### PR DESCRIPTION
**Pull request by maltemorgenstern**
## Description
Fixes a few typos and broken markdown links in the documentation.

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.

### CODE REVIEW

File: docs/docs/compliance/compliance.md [hunk 1]
```diff
@@ -26,9 +26,11 @@ Once the report has been generated, you can fetch and review its results section
 kubectl get compliance nsa  -o=jsonpath='{.status}' | jq .
 ```
 
-## You can create your own custom compliance report. A compliance report is a simple YAML document in the following format:
+## Custom compliance reports
 
-<summary>customer compliance report</summary>
+You can create your own custom compliance report. A compliance report is a simple YAML document in the following format:
+
+<summary>Customer compliance report</summary>
```
Comment (chen-keinan) [inline docs/docs/compliance/compliance.md:33]: I think we can remove this : `<summary>Customer compliance report</summary>` as well

**Review by chen-keinan — COMMENTED** [review]
@maltemorgenstern Thanks for catching up all this typo issues. lgtm 🚀  I have added one comment.

fyi: you can test docs locally by running `mage docs:serve`

File: docs/docs/compliance/compliance.md (hunk 1 above)
Comment (maltemorgenstern) [reply docs/docs/compliance/compliance.md]: Done.

**Comment by maltemorgenstern** [comment]
> fyi: you can test docs locally by running `mage docs:serve`

Good to know, thanks for the hint!

**Review by chen-keinan — APPROVED** [review]

**Pull request merged**